FeatureRequest: Enable Pandas `options.mode.copy_on_write = True` Compatibility

### Feature type

- [ ] Add new functionality

- [X] Change existing functionality


### General description of the proposed functionality

I'm opening this as a feature request for now but this may become a bug as later versions of Pandas (namely, `pandas >= 3.x`) are used by Pycytominer. In reading the [release notes for Pandas 2.2.0](https://pandas.pydata.org/pandas-docs/version/2.2.0/whatsnew/v2.2.0.html#copy-on-write), I noticed the following:

> The currently optional mode Copy-on-Write will be enabled by default in pandas 3.0. There won’t be an option to keep the current behavior enabled.

This will change the behavior and what's happens with Pandas DataFrames and Series when it comes to subset mutations ([see here for more](https://pandas.pydata.org/pandas-docs/version/2.2.0/user_guide/copy_on_write.html#copy-on-write)). My understanding from the above release documentation is that there may be no flexibility to retain the existing default functionality. Based on this, Pycytominer could add in the incoming default and address any changes in advance to ready for Pandas 3.0. As the Pandas docs mention as well, "[With Copy-on-Write] the average performance and memory usage will improve." ([link](https://pandas.pydata.org/pandas-docs/version/2.2.0/user_guide/copy_on_write.html#copy-on-write)).

I experimented with this quickly with Python 3.11.2 ([see here](https://github.com/d33bs/pycytominer/tree/pandas-copy_on_write-testing)) and found there were some failing tests:

```log
FAILED tests/test_consensus.py::test_consensus_modz - ValueError: underlying array is read-only
FAILED tests/test_cyto_utils/test_modz.py::test_modz_base - ValueError: underlying array is read-only
FAILED tests/test_cyto_utils/test_modz.py::test_modz - ValueError: underlying array is read-only
FAILED tests/test_cyto_utils/test_modz.py::test_modz_extraneous_column - ValueError: underlying array is read-only
FAILED tests/test_cyto_utils/test_modz.py::test_modz_multiple_columns - ValueError: underlying array is read-only
FAILED tests/test_cyto_utils/test_modz.py::test_modz_multiple_columns_one_metadata_column - ValueError: underlying array is read-only
FAILED tests/test_cyto_utils/test_modz.py::test_modz_multiple_columns_feature_specify - ValueError: underlying array is read-only
FAILED tests/test_cyto_utils/test_modz.py::test_modz_unbalanced_sample_numbers - ValueError: underlying array is read-only
```


### Feature example

To enable this, I added a setting to the root Pycytominer `__init__.py` but there may be a more appropriate place to set this.

### Alternative Solutions

I'm unsure of the timeline for Pandas 3.0, and Pandas 2.0 deprecation. That said, we likely have some time to adjust for this.

### Additional information

_No response_

--- a/pycytominer/__config__.py
+++ b/pycytominer/__config__.py
@@ -1,0 +1,9 @@
+"""
+Module used for pycytominer configuration details.
+"""
+
+import pandas as pd
+
+# configure pandas copy_on_write for 3.0.0 requirements
+# see: https://pandas.pydata.org/pandas-docs/version/2.2.0/whatsnew/v2.2.0.html#copy-on-write
+pd.options.mode.copy_on_write = True

--- a/pycytominer/__init__.py
+++ b/pycytominer/__init__.py
@@ -1,4 +1,4 @@
-from pycytominer import __about__
+from pycytominer import __about__, __config__
 
 from .aggregate import aggregate
 from .annotate import annotate

--- a/pycytominer/cyto_utils/modz.py
+++ b/pycytominer/cyto_utils/modz.py
@@ -1,4 +1,5 @@
 import numpy as np
+import pandas as pd
 from pycytominer.cyto_utils.util import (
     get_pairwise_correlation,
     check_correlation_method,
@@ -43,9 +44,17 @@ def modz_base(population_df, method="spearman", min_weight=0.01, precision=4):
     # Round correlation results
     pair_df = pair_df.round(precision)
 
+    # create a copy of cor_df values for use with np.fill_diagonal
+    cor_df_values = cor_df.values.copy()
+
     # Step 2: Identify sample weights
     # Fill diagonal of correlation_matrix with np.nan
-    np.fill_diagonal(cor_df.values, np.nan)
+    np.fill_diagonal(cor_df_values, np.nan)
+
+    # reconstitute the changed data as a new dataframe to avoid read-only behavior
+    cor_df = pd.DataFrame(
+        data=cor_df_values, index=cor_df.index, columns=cor_df.columns
+    )
 
     # Remove negative values
     cor_df = cor_df.clip(lower=0)

